src/components/network/line-layer.js

```diff
@@ -195,336 +206,808 @@ class LineLayer extends CompositeLayer {
+                        getLinePositions: (line) => line.positions,
+                        getColor: (arrow) =>
+                            getLineColor(
+                                arrow.line,
+                                nominalVoltageColor,
+                                props,
+                                state.linesConnection.get(arrow.line.id)
+                            ),
+                        getSize: 700,
+                        getSpeedFactor: (arrow) =>
+                            getArrowSpeedFactor(getArrowSpeed(arrow.line)),
+                        getLineParallelIndex: (arrow) =>
+                            arrow.line.geo.parallelIndex,
+                        getLineAngles: (arrow) => [
+                            arrow.line.geo.angleStart,
+                            arrow.line.geo.angle,
+                            arrow.line.geo.angleEnd,
+                        ],
+                        getProximityFactors: (arrow) => [
+                            arrow.line.geo.proximityFactorStart,
+                            arrow.line.geo.proximityFactorEnd,
+                        ],
+                        getDistanceBetweenLines: props.distanceBetweenLines,
+                        maxParallelOffset: props.maxParallelOffset,
+                        minParallelOffset: props.minParallelOffset,
+                        getDirection: (arrow) => {
+                            return getArrowDirection(arrow.line.p1);
+                        },
+                        animated:
+                            props.showLineFlow &&
+                            props.lineFlowMode === LineFlowMode.ANIMATED_ARROWS,
+                        visible:
+                            props.showLineFlow &&
+                            props.filteredNominalVoltages.includes(
+                                nominalVoltage
+                            ) &&
+                            state[PATH_UPDATED + nominalVoltageIndex] <=
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                        updateTriggers: {
+                            getLinePositions: [
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                            ],
+                            getLineParallelIndex: [
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                            ],
+                            getLineAngles: [
+                                state[ARROW_UPDATE + nominalVoltageIndex],
+                            ],
+                            getColor: [
+                                props.lineFlowColorMode,
+                                props.lineFlowAlertThreshold,
+                                props.updatedLines,
+                            ],
+                        },
+                    })
+                )
+        );
+        this.setState({ layers: layers });
+        let next = undefined;
```

Comment:
why `= undefined`

Reply:
long habit of never declare uninitialized variables